forge-gui/res/cardsfolder/upcoming/meticulous_excavation.txt

```diff
@@ -0,0 +1,10 @@
+Name:Meticulous Excavation
+ManaCost:W
+Types:Enchantment
+A:AB$ ChangeZone | Cost$ 2 W | RememberTargets$ True | PlayerTurn$ True | ValidTgts$ Permanent.YouCtrl | TgtPrompt$ Select target permanent you control | SubAbility$ ExileWithUnearth | Origin$ Battlefield | Destination$ Hand | SpellDescription$ Return target permanent you control to its owner's hand. If it has unearth, instead exile it, then return that card to its owner's hand. Activate only during your turn.
+SVar:DBEffect:DB$ Effect | RememberObjects$ Remembered | ConditionDefined Remembered | ConditionPresent$ Card.withUnearth | ExileOnMoved$ Stack | ReplacementEffects$ ReplaceGraveyard | SubAbility$ DBCleanup
```

Comment:
This looks very weird, was there a template you used?
It shouldn't even work because the Stack plays no real role there

Reply:
Yeah this smells like you want AB$ Branch